src/GafferArnold/IECoreArnoldPreview/ShaderNetworkAlgo.cpp

```diff
@@ -0,0 +1,243 @@
+//////////////////////////////////////////////////////////////////////////
```

Comment:
Fair point. For the record, this file contains the most substantial changes of all in this PR - it's _much_ simpler now. Fortunately, it's changed enough that I think reviewing it as new code rather than as diffs is probably the most effective approach anyway.